### PR TITLE
feat: allow concepts to be associated with `data-dfn-for`

### LINKED ISSUE #2271: Let definitions in specs be "for" particular things
For WebIDL, we have the ability to say that the definition of `attribute X` is for `SomeInterface`. It would be nice to be able to expand this concept globally so that:

 * A definition can be for a particular thing (e.g., "sort" is for "list").
 * A internal slot is for an interface. (e.g., we can then have: [[request]] is for PaymentRequest, [[request]] is for PaymentResponse and a way to define both, but without them conflicting)

### PR DESCRIPTION
**Pull request by sidvishnoi**
Closes #2271
Cleanup and rework of #2508

### CODE REVIEW

File: src/core/inline-idl-parser.js [hunk 1]
```diff
@@ -141,7 +141,9 @@ function renderBase(details) {
   // Check if base is a local variable in a section
   const { identifier, renderParent } = details;
   if (renderParent) {
-    return html`<a data-xref-type="_IDL_"><code>${identifier}</code></a>`;
+    return html`<a data-xref-type="_IDL_" data-link-type="idl"
```
Comment (sidvishnoi) [inline src/core/inline-idl-parser.js:144]: This sends this PR into a sad state..

File: src/core/link-to-dfn.js [hunk 2]
```diff
@@ -108,7 +108,10 @@ function collectDfns(title) {
   const result = new Map();
   const duplicates = [];
   for (const dfn of definitionMap.get(title)) {
-    const { dfnFor = "", dfnType = "dfn" } = dfn.dataset;
+    const { dfnType = "dfn" } = dfn.dataset;
```
Comment (sidvishnoi) [inline src/core/link-to-dfn.js:111]: This broke a lot of specs :disappointed:

**Comment by marcoscaceres** [comment]
@sidvishnoi, try to land tiny bits if you can... go for quick wins (e.g., the src/core/inline-idl-parser.js annotations). It might make this a little less overwhelming to code and review.

File: src/core/link-to-dfn.js [hunk 3]
```diff
@@ -169,8 +169,12 @@ function findMatchingDfn(anchor, titleToDfns) {
 function processAnchor(anchor, dfn, titleToDfns) {
   let noLocalMatch = false;
   const { linkFor } = anchor.dataset;
+  const { dfnFor } = dfn.dataset;
   if (dfn.dataset.cite) {
     anchor.dataset.cite = dfn.dataset.cite;
+  } else if (linkFor === dfnFor) {
```
Comment (sidvishnoi) [inline src/core/link-to-dfn.js]: This happens when the "dfnFor" is defined externally, so `titleToDfns` doesn't have dfnFor/linkFor.

**Comment by sidvishnoi** [comment]
Closed in favor of https://github.com/w3c/respec/pull/2873

**Pull request closed without merging**